admin/services/restart.service.js

```diff
@@ -110,4 +111,47 @@ restartService.canRestart = async pupilId => {
   return hasUsedRestart && hasRestartAttemptRemaining && hasCheckAttemptRemaining
 }
 
+/**
+ * Get pupils who have been submitted for a restart
+ * @param schoolId
+ * @returns {Array}
+ */
+
+restartService.getSubmittedRestarts = async schoolId => {
+  let pupils = await pupilDataService.getSortedPupils(schoolId, 'lastName', 'asc')
+  if (!pupils || pupils.length === 0) return []
+  let restarts = []
+  await Promise.all(pupils.map(async p => {
+    const r = await pupilRestartDataService.findOne({ pupilId: p._id, isDeleted: false })
+    if (r) {
+      const status = await restartService.getStatus(p._id)
+      restarts.push({
+        _id: r._id,
+        reason: r.reason,
+        status,
+        foreName: p.foreName,
+        lastName: p.lastName,
+        middleNames: p.middleNames,
+        dob: dateService.formatShortGdsDate(p.dob)
+      })
+    }
+  }))
+  restarts = pupilIdentificationFlagService.addIdentificationFlags(restarts)
+  return restarts
+}
+
+/**
+ * Get pupil's restart status
+ * @param pupilId
+ * @returns {String}
+ */
+
+restartService.getStatus = async pupilId => {
```

Comment:
The service here seems to know a lot about the UI - which is where I presume the message are going.  I would suggest that if we must provide UI elements from a business service (policy) then we should provide two elements:  first a status code, and second the user-friendly message.

@GuyHarwood - what do you think?

Reply:
it does feel like a UI requirement is added at the logic layer. 

although I wonder how much value-add we will get from refactoring it 🤔

on one side it's 'just enough to meet the requirements' and on the other its fulfilling UI reqs at the logic layer.